core/fluxsync/adapters.go

```diff
@@ -294,14 +303,58 @@ func (obj ImageUpdateAutomationAdapter) GroupVersionKind() schema.GroupVersionKi
 	return imgautomationv1.GroupVersion.WithKind(imgautomationv1.ImageUpdateAutomationKind)
 }
 
-func (obj ImageUpdateAutomationAdapter) SetSuspended(suspend bool) {
+func (obj ImageUpdateAutomationAdapter) SetSuspended(suspend bool) error {
 	obj.Spec.Suspend = suspend
+	return nil
 }
 
 func (obj ImageUpdateAutomationAdapter) DeepCopyClientObject() client.Object {
 	return obj.DeepCopy()
 }
 
+type UnstructuredAdapter struct {
+	*unstructured.Unstructured
+}
+
+func (obj UnstructuredAdapter) GetLastHandledReconcileRequest() string {
+	if val, found, _ := unstructured.NestedString(obj.Object, "status", "lastHandledReconcileAt"); found {
+		return val
+	}
+	return ""
+}
+
+func (obj UnstructuredAdapter) GetConditions() []metav1.Condition {
+	conditionsSlice, found, err := unstructured.NestedSlice(obj.Object, "status", "conditions")
+	if !found || err != nil {
+		return nil
+	}
+
+	var conditions []metav1.Condition
+	for _, c := range conditionsSlice {
+		var condition metav1.Condition
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(c.(map[string]interface{}), &condition); err != nil {
+			continue
+		}
+		conditions = append(conditions, condition)
+	}
+
+	return conditions
+}
+
+func (obj UnstructuredAdapter) AsClientObject() client.Object {
+	// Important for the client reflection stuff to work
+	// We can't return just `obj` here as it seems to break stuff.
+	return obj.Unstructured
```

Comment:
I tried this in tests, and they don't fail when this is just `return obj` ?

Reply:
Great catch, updated the test here to cover that case my bad. Seems to pop up during get/list not create.